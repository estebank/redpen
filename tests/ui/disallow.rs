#[redpen::disallow(T = "Pineapple")]
#[redpen::disallow(T = "&Pineapple")]
#[redpen::disallow(T = "std::string::String")]
struct Pizza<T>(T);

struct Pineapple;

type Alias = Pizza<Pineapple>; //~ ERROR
type Alias2 = Pizza<String>; //~ ERROR

fn foo(p: Pizza<Pineapple>) -> Pizza<Pineapple> {
    p
}

fn main() {
    let _ = Pizza(Pineapple);
    let _ = Pizza(String::new());

    let _ = &vec!(Pizza(&Pizza(Pineapple)));
    let _ = &vec!(Pizza(&Pizza(&Pineapple)));
}