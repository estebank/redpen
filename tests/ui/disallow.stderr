error: type parameter `T` in `Pizza` can't be `Pineapple`
 --> $DIR/disallow.rs:8:20
  |
8 | type Alias = Pizza<Pineapple>; //~ ERROR
  |                    ^^^^^^^^^
  |
  = note: `#[deny(redpen::disallow)]` on by default

error: type parameter `T` in `Pizza` can't be `std::string::String`
 --> $DIR/disallow.rs:9:21
  |
9 | type Alias2 = Pizza<String>; //~ ERROR
  |                     ^^^^^^

error: type parameter `T` in `Pizza` can't be `Pineapple`
  --> $DIR/disallow.rs:11:17
   |
11 | fn foo(p: Pizza<Pineapple>) -> Pizza<Pineapple> {
   |                 ^^^^^^^^^

error: type parameter `T` in `Pizza` can't be `Pineapple`
  --> $DIR/disallow.rs:11:38
   |
11 | fn foo(p: Pizza<Pineapple>) -> Pizza<Pineapple> {
   |                                      ^^^^^^^^^

error: type parameter `T` in `Pizza` can't be `Pineapple`
  --> $DIR/disallow.rs:12:5
   |
12 |     p
   |     ^ this expression is of type `Pizza<Pineapple>`

error: type parameter `T` in `Pizza` can't be `Pineapple`
  --> $DIR/disallow.rs:16:13
   |
16 |     let _ = Pizza(Pineapple);
   |             ^^^^^^^^^^^^^^^^ this expression is of type `Pizza<Pineapple>`

error: type parameter `T` in `Pizza` can't be `std::string::String`
  --> $DIR/disallow.rs:17:13
   |
17 |     let _ = Pizza(String::new());
   |             ^^^^^^^^^^^^^^^^^^^^ this expression is of type `Pizza<std::string::String>`

error: type parameter `T` in `Pizza` can't be `Pineapple`
  --> $DIR/disallow.rs:19:26
   |
19 |     let _ = &vec!(Pizza(&Pizza(Pineapple)));
   |                          ^^^^^^^^^^^^^^^^ this expression is of type `Pizza<Pineapple>`

error: type parameter `T` in `Pizza` can't be `&Pineapple`
  --> $DIR/disallow.rs:20:26
   |
20 |     let _ = &vec!(Pizza(&Pizza(&Pineapple)));
   |                          ^^^^^^^^^^^^^^^^^ this expression is of type `Pizza<&Pineapple>`

error: aborting due to 9 previous errors

